### PR TITLE
[API] Adding shipping methods

### PR DESCRIPTION
**Pull request by lchrusciel**
| Q               | A
| --------------- | -----
| Branch?         | master
| Bug fix?        | no
| New feature?    | yes
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | part of #11250 
| License         | MIT

### CODE REVIEW

File: src/Sylius/Behat/Context/Transform/ZoneContext.php [hunk 1]
```diff
@@ -63,7 +63,7 @@ public function getRestOfTheWorldZone()
     /**
      * @return ZoneInterface
      */
```
Comment (AdamKasp) [inline src/Sylius/Behat/Context/Transform/ZoneContext.php]:
```suggestion

```

Comment (lchrusciel) [reply src/Sylius/Behat/Context/Transform/ZoneContext.php]: Will be covered in the next PR

File: src/Sylius/Behat/Client/ResponseChecker.php [hunk 2]
```diff
@@ -89,11 +89,8 @@ public function hasItemOnPositionWithValue(Response $response, int $position, st
     public function hasItemWithTranslation(Response $response, string $locale, string $key, string $translation): bool
     {
         foreach ($this->getCollection($response) as $resource) {
-            if (
-                isset($resource['translations']) &&
-                isset($resource['translations'][$locale]) &&
-                $resource['translations'][$locale][$key] === $translation
-            ) {
+            if (isset($resource['translations'][$locale]) && $resource['translations'][$locale][$key] === $translation)
+            {
```
Comment (GSadee) [inline src/Sylius/Behat/Client/ResponseChecker.php]: Should be in line before

File: src/Sylius/Behat/Context/Api/Admin/ManagingShippingMethodsContext.php [hunk 3]
```diff
@@ -50,6 +59,73 @@ public function iDeleteShippingMethod(ShippingMethodInterface $shippingMethod):
...
+    /**
+     * @When I specify its position as :position
+     */
+    public function iSpecifyItsPositionAs(int $position): void
+    {
+        $this->client->addRequestData('position', $position);
+    }
+
+    /**
+     * @When I name it :name in :language
```
Comment (GSadee) [inline src/Sylius/Behat/Context/Api/Admin/ManagingShippingMethodsContext.php]:
```suggestion
     * @When I name it :name in :localeCode
```

File: src/Sylius/Behat/Context/Api/Admin/ManagingShippingMethodsContext.php [hunk 4]
```diff
@@ -88,7 +165,18 @@ public function thisShippingMethodShouldNoLongerExistInTheRegistry(ShippingMetho
...
+    }
+
+    /**
+     * @Then I should be notified that it has been successfully created
+     */
+    public function iShouldBeNotifiedThatItHasBeenSuccessfullyCreated(): void
+    {
+        Assert::true(
+            $this->responseChecker->isCreationSuccessful($this->client->getLastResponse()),
+            'Zone could not be created'
```
Comment (GSadee) [inline src/Sylius/Behat/Context/Api/Admin/ManagingShippingMethodsContext.php]: Zone?

**Comment by GSadee** [comment]
Thanks, Łukasz! :1st_place_medal:

File: src/Sylius/Behat/Context/Api/Admin/ManagingShippingMethodsContext.php (hunk 3 above)
Comment (lchrusciel) [reply src/Sylius/Behat/Context/Api/Admin/ManagingShippingMethodsContext.php]: This value is transformed, I cannot change it to localeCode I suppose

**Pull request merged**